### PR TITLE
Update the docker file to make it smaller and change the role created

### PR DESCRIPTION
**Pull request by joaopapereira**

### CODE REVIEW

**Comment by joaopapereira** [comment]
This PR is the followup of PR #4404 with:
- Smaller image size
- Change the role from `gp` to `gpadmin`
- Remove libraries that are not needed

**Comment by Eulerizeit** [comment]
@kmacoskey @pivotal-mike we would eventually like to add something like this to either the release or toolsmiths pipeline so that it can be leveraged in a pgAdmin CI run by the pgAdmin community.

File: src/tools/docker/ubuntu-16.04/Dockerfile [hunk 1]
```diff
@@ -1,58 +1,46 @@
...
+ && sed -i 's/gpdata1/gpdata\/gpdata1/g' /gpdata/gpinitsystem_singlenode \
+ && sed -i 's/gpdata2/gpdata\/gpdata2/g' /gpdata/gpinitsystem_singlenode \
+ && sed -i 's/gpmaster/gpdata\/gpmaster/g' /gpdata/gpinitsystem_singlenode
+
+# Create gpadmin user and add the user to the sudoers
+RUN useradd -md /home/gpadmin/ gpadmin \
+ && chown gpadmin -R /gpdata \
+ && echo "source /opt/gpdb/greenplum_path.sh" > /home/gpadmin/.bash_profile && chown gpadmin:gpadmin /home/gpadmin/.bash_profile \
+ && su - gpadmin bash -c 'mkdir /home/gpadmin/.ssh' \
+ && echo "gpadmin ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers \
```
Comment (kmacoskey) [inline src/tools/docker/ubuntu-16.04/Dockerfile]: Is this necessary or just convenient?

File: src/tools/docker/ubuntu-16.04/Dockerfile [hunk 2]
```diff
@@ -1,58 +1,46 @@
...
+# Install GreenPlum
+RUN wget -qO- https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+ && add-apt-repository ppa:greenplum/db \
+ && apt update \
+ && apt install -y greenplum-db-oss \
+ && locale-gen en_US.utf8 
+
+# Create GreenPlum data folders and copy/edit configuration to use a single node
+RUN mkdir /gpdata \
+ && mkdir /gpdata/gpdata1 \
```
Comment (kmacoskey) [inline src/tools/docker/ubuntu-16.04/Dockerfile]: Any specific reason `/gpdata` is necessary? The more common convention would be `/data`

**Comment by Eulerizeit** [comment]
It is expected to work for 5.x we are just pushing the PR through master. For the Ubuntu specific stuff that might be a little weird.

**Comment by kmacoskey** [comment]
That is a little weird. I suggest we call the image `pivotaldata/gpdb5:latest` in order to delineate. That would be a couple of small changes throughout the PR.

File: src/tools/docker/ubuntu-16.04/Dockerfile (hunk 1 above)
Comment (joaopapereira) [reply src/tools/docker/ubuntu-16.04/Dockerfile]: This is the result of some iterations, but I believe we need gpadmin to be a sudoer

File: src/tools/docker/ubuntu-16.04/Dockerfile (hunk 2 above)
Comment (joaopapereira) [reply src/tools/docker/ubuntu-16.04/Dockerfile]: There is no particular reason for this, was just following the blog post on how to install gpdb in ubuntu. But I will change this

**Comment by joaopapereira** [comment]
@kmacoskey Just updated the PR with all your suggestions

File: src/tools/docker/ubuntu-16.04/Dockerfile [hunk 3]
```diff
@@ -1,58 +1,46 @@
-# currently published as pivotaldata/pgadmin:gpdb-5
+# currently published as pivotaldata/pgadmin:latest
```
Comment (kmacoskey) [inline src/tools/docker/ubuntu-16.04/Dockerfile]: The comment is incorrect? 

To make sure I have it straight:
`pivotaldata/gpdb5:latest` will be a ubuntu-16.04 image with the latest gpdb5X OSS installed. The Dockerfile is included in `gpdb` source itself and we will want a new process somewhere that will create and push to `pivotaldata` for new tags on `5X_STABLE`. 

`pivotaldata/pgadmin:latest` will eventually be built on top of `pivotaldata/gpdb5:latest` and will include the latest pgadmin. The Dockerfile will live somewhere not in `gpdb` source and we will want a new process somewhere that will create and push to `pivotaldata` for new combinations of a tag on `5X_STABLE` and tag on `pgadmin` (this part is a little fuzzy)

Is this correct?

If so, then yes the comment should be 
```
# currently published as pivotaldata/gpdb5:latest
```

If not, then lets chat some more.

**Comment by joaopapereira** [comment]
Closed with commit: 3183577d44b

**Pull request closed without merging**